inc/uninstall.class.php

```diff
@@ -500,6 +503,141 @@ static function deleteFusionInventoryLink($itemtype, $items_id) {
+               "itemtype" => "NetworkPort",
+               "items_id" => $networkport_id,
+               'is_dynamic' => 1
+            ]
+         );
+      }
+
+      // unlock item relations
+      $RELATION = getDbRelations();
+      if (isset($RELATION[$item->getTable()])) {
+         foreach ($RELATION[$item->getTable()] as $tablename => $fields) {
+            if ($tablename[0] == '_') {
+               $tablename = ltrim($tablename, '_');
+            }
+
+            $sub_itemtype = getItemTypeForTable($tablename);
+            $sub_item = getItemForItemtype($sub_itemtype);
+
+            if ($sub_item === false || !$sub_item->maybeDynamic()) {
+               continue;
+            }
+
+            if (in_array(get_class($sub_item), [Agent::class, Computer_Item::class, Lockedfield::class, NetworkName::class])) {
+               // Specific handling
+               continue;
+            }
+
+            foreach ($fields as $field) {
+               if (is_array($field)) {
+                  // Relation based on 'itemtype'/'items_id' (polymorphic relationship)
+                  if ($item instanceof IPAddress && in_array('mainitemtype', $field) && in_array('mainitems_id', $field)) {
+                     // glpi_ipaddresses relationship that does not respect naming conventions
+                     $itemtype_field = 'mainitemtype';
+                     $items_id_field = 'mainitems_id';
+                  } else {
+                     $itemtype_matches = preg_grep('/^itemtype/', $field);
+                     $items_id_matches = preg_grep('/^items_id/', $field);
+                     $itemtype_field = reset($itemtype_matches);
+                     $items_id_field = reset($items_id_matches);
+                  }
+                  $DB->update(
+                     $tablename,
+                     [
+                        'is_deleted' => 0,
+                        'is_dynamic' => 0
+                     ],
+                     [
+                        $items_id_field => $item->getID(),
+                        $itemtype_field => $item->getType(),
+                        'is_dynamic' => 1
+                     ]
+                  );
+               } else {
+                  // Relation based on single foreign key
+                  $DB->update(
+                     $tablename,
+                     [
+                        'is_deleted' => 0,
+                        'is_dynamic' => 0
+                     ],
+                     [
+                        $field => $item->getID(),
+                        'is_dynamic' => 1
+                     ]
+                  );
+               }
+            }
+         }
+      }
+
+      //remove is_dynamic from asset
+      if ($item->maybeDynamic()) {
+         $DB->update(
+            Computer::getTable(),
+            ['is_dynamic' => false],
+            ['id' => $item->getID()]
+         );
+         //reload
+         $item->getFromDB($item->getID());
+      }
+   }
+
```

Comment:
Some data / link are not purged

![image](https://github.com/pluginsGLPI/uninstall/assets/7335054/5a831d22-7edc-4053-8bc2-6820625f6e3e)